src/Altinn.App.Api/Controllers/PersonSearchController.cs

```diff
@@ -0,0 +1,54 @@
+using System.Net.Mime;
+using Altinn.App.Api.Infrastructure.Filters;
+using Altinn.App.Api.Models;
+using Altinn.App.Core.Internal.Registers;
+using Altinn.Platform.Register.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Altinn.App.Api.Controllers;
+
+/// <summary>
+/// This controller class provides Folkeregisteret (DSF) person search functionality.
+/// </summary>
+[AutoValidateAntiforgeryTokenIfAuthCookie]
+[ApiController]
+[Produces(MediaTypeNames.Application.Json)]
+[Consumes(MediaTypeNames.Application.Json)]
+[ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
+[Route("{org}/{app}/api/v1/person-search")]
+public class PersonSearchControllerController : ControllerBase
+{
+    private readonly IPersonClient _personClient;
+
+    /// <summary>
+    /// Initialize a new instance of <see cref="PersonSearchControllerController"/> with the given services.
+    /// </summary>
+    /// <param name="personClient">A client for searching for a person.</param>
+    public PersonSearchControllerController(IPersonClient personClient)
+    {
+        _personClient = personClient;
+    }
+
+    /// <summary>
+    /// Allows searching for a person in Folkeregisteret (DSF)
+    /// </summary>
+    /// <param name="personSearchRequest">Payload that contains params for executing a person search.</param>
+    /// <param name="cancellationToken"></param>
+    /// <returns>A <see cref="PersonSearchResponse"/> object.</returns>
+    [HttpPost]
+    [ProducesResponseType(typeof(PersonSearchResponse), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<PersonSearchResponse>> SearchForPerson(
+        [FromBody] PersonSearchRequest personSearchRequest,
+        CancellationToken cancellationToken
+    )
+    {
+        Person? person = await _personClient.GetPerson(
+            personSearchRequest.SocialSecurityNumber,
+            personSearchRequest.LastName,
+            cancellationToken
+        );
+
+        return person is null ? NotFound() : PersonSearchResponse.MapFromPerson(person);
```

Comment:
Is 404 the correct status code when a search returns no results? Isn't it better to have 200 and modify `PersonSearchResponse` to have a `Success` property. The code is mostly for looking at logs, and a miss is a valid result and no reason to investigate.

Somehow Swagger seems to think that you return `ProblemDetails` for the 404, so if we keep the code, it need to actually return what we promised.

Reply:
alternatively 204 No Content

Reply:
@ivarne Take another look.